src/main/java/com/teragrep/pth10/ast/bo/ElementNode.java

```
@@ -57,51 +58,30 @@
 
-    Element val = null;
-
-    public ElementNode(Token token) {
-        super(token);
-    }
+    private final Element element;
 
     public ElementNode(Element element) {
-        this.val = element;
+        this.element = element;
     }
 
     public Element getElement() {
-        return val;
+        return element;
     }
 
+    @Override
     public String toString() {
-        String str = null;
-        try {
-            TransformerFactory transFactory = TransformerFactory.newInstance();
-            Transformer transformer = transFactory.newTransformer();
-            StringWriter buffer = new StringWriter();
-            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-            transformer.transform(new DOMSource(val), new StreamResult(buffer));
-            str = buffer.toString();
-        }
-        catch (TransformerConfigurationException tex) {
-        }
-        catch (TransformerException ex) {
-        }
-        return str;
-    }
-
-    public static String toString(Element val) {
-        String str = null;
+        final StringWriter buffer = new StringWriter();
         try {
-            TransformerFactory transFactory = TransformerFactory.newInstance();
-            Transformer transformer = transFactory.newTransformer();
-            StringWriter buffer = new StringWriter();
+            final TransformerFactory transFactory = TransformerFactory.newInstance();
+            transFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            final Transformer transformer = transFactory.newTransformer();
             transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-            transformer.transform(new DOMSource(val), new StreamResult(buffer));
-            str = buffer.toString();
+            transformer.transform(new DOMSource(element), new StreamResult(buffer));
         }
         catch (TransformerConfigurationException tex) {
```

I think these should throw an exception

---

caught exceptions rethrown